### PR TITLE
Fix batched ack tracker logsource ref

### PR DESCRIPTION
**Pull request by lbudai**

### CODE REVIEW

**Comment by lbudai** [comment]
@MrAnno : could you check it?
and many thanks for helping during the investigation!!

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: lib/ack-tracker/batched_ack_tracker.c [hunk 1]
```diff
@@ -201,16 +199,76 @@ _stop_watches(BatchedAckTracker *self)
...
-      iv_event_unregister(&self->request_restart_timer);
       _stop_batch_timer(self);
       self->watches_running = FALSE;
     }
 }
 
+static void
+__free(AckTracker *s)
+{
+  msg_debug("BatchedAckTracker::free");
```
Comment (MrAnno) [inline lib/ack-tracker/batched_ack_tracker.c:210]: Do we want to move these messages to the trace level?

**Review by MrAnno — APPROVED** [review]
This is a difficult one :) 👍🏻

**Comment by MrAnno** [comment]
<details>
  <summary>Reproducer for the crash (click to expand):</summary>

```
from syslogng import LogSource
from syslogng import LogMessage
from syslogng import Logger
from syslogng import BatchedAckTracker

import signal
signal.signal(signal.SIGINT, signal.SIG_IGN)

import threading
import time

class TestSource(LogSource):
    def init(self, options):
        self.stop_event = threading.Event()

        self.ack_tracker = BatchedAckTracker(timeout=5000, batch_size=100,
                                             batched_ack_callback=self.save_bookmarks)
        return True

    def deinit(self):
        pass

    def save_bookmarks(self, bookmarks):
        for b in bookmarks:
            print(b)

    def run(self):
        while not self.stop_event.is_set():
            msg = LogMessage("{}", bookmark="bookmark-data")
            # msg.set_bookmark("bookmark-data")
            self.post_message(msg)

    def request_exit(self):
        self.stop_event.set()

```

```
@version: 3.28
@include "scl.conf"

log {
  source {
    python(class("testsource.TestSource"));
  };

  destination {
    network("127.0.0.1" port(4444));
  };

  flags(flow-control);
};
```

</details>

**Review by gaborznagy — APPROVED** [review]

**Review by MrAnno — APPROVED** [review]

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by lbudai** [comment]
I'm merging this as the change was only change msg_debug to msg_trace and before that the PR got two +1.

**Pull request merged**